connector/src/main/java/com/google/cloud/spark/bigquery/v2/BigQueryDataSourceWriterModule.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery.v2;
+
+import com.google.cloud.bigquery.connector.common.BigQueryClient;
+import com.google.cloud.spark.bigquery.SparkBigQueryConfig;
+import com.google.common.base.Preconditions;
+import com.google.inject.Binder;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.SaveMode;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.StructType;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Optional;
+import java.util.UUID;
+
+public class BigQueryDataSourceWriterModule implements Module {
+
+  private final String writeUUID;
+  private final StructType sparkSchema;
+  private final SaveMode mode;
+
+  BigQueryDataSourceWriterModule(String writeUUID, StructType sparkSchema, SaveMode mode) {
+    this.writeUUID = writeUUID;
+    this.sparkSchema = sparkSchema;
+    this.mode = mode;
+  }
+
+  @Override
+  public void configure(Binder binder) {
+    // empty
+  }
+
+  @Singleton
```

Comment:
The same question.

Reply:
The `BigQueryDataSourceWriter` is created once per write operation, so it is singleton within this context